app/jobs/screenshot_workspace_job.rb

```diff
@@ -18,17 +21,17 @@ def perform(workspace, timestamp = nil)
       end
     end
   end
+  # rubocop:enable Metrics/AbcSize,Metrics/MethodLength
 
-  def embed_path_with_token(workspace, timestamp)
-    secret = JWT.encode(
-      { workspace: workspace.to_param, exp: 5.minutes.from_now.to_i },
-      Rails.application.secret_key_base,
-      'HS256'
-    )
+  private
 
-    Rails.application.routes.url_helpers.url_for(
-      { action: 'embed', controller: 'workspaces', id: workspace.to_param, token: secret, timestamp: timestamp }
-        .merge(Settings.screenshot_url_parameters.to_h)
-    )
+  def token
+    JWT.encode({ exp: 5.minutes.from_now.to_i },
```

Comment:
If we continue to include the workspace in this token, we can use that data over in Ability to restrict access to just that workspace/project (instead of just letting a token-bearing user read anything they want 🤷 )

Reply:
I thought about that, and I also thought, "no real users have tokens, just puppeteers." Perhaps that is short-sighted.